.github/workflows/validate-theme-entry.yml

```diff
@@ -0,0 +1,263 @@
+                    if (!validKeys.includes(key) && !key === 'version') {
+                        addError(`Your manifest has the invalid \`${key}\` property.`);
+                    }
+                }
+
+                for (let key in validManifestKeys) {
+                    if (!manifest.hasOwnProperty(key)) {
+                        addError(`Your manifest does not have the required \`${key}\` property.`);
+                    }
+                }
+
+                if (manifest.hasOwnProperty('name') && manifest.name != theme.name) {
+                    addError(`Plugin name mismatch, the name in this PR (\`${theme.name}\`) is not the same as the one in your repo (\`${manifest.name}\`). If you just changed your plugin name, remember to change it in the manifest.json in your repo and your latest GitHub release.`);
+                }
+
+                if (manifest.hasOwnProperty('version') && !(/^[0-9.]+$/i.test(manifest.version))) {
+                    addError('Your latest version number is not valid. Only numbers and dots are allowed.');
+                }
+
+                if (manifest.hasOwnProperty('author') && /^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$/.test(manifest.author)) {
+                    addWarning('We generally discourage from including email addresses in the `author` field.');
+                }
+
+
+                try {
+                    await github.rest.repos.getContent({
+                        owner, repo, path: 'theme.css'
+                    });
+                } catch (e) {
+                    addError('Your repository does not include a `theme.css` file');
+                }
+
+                try {
+                    await github.rest.repos.getContent({
+                        owner, repo, path: 'obsidian.css'
+                    });
+                    addWarning('Your repository includes a `obsidian.css` file, this is only used in legacy versions of Obsidian.');
+                } catch (e) { }
+
+                try {
+                    const screenshot = await github.rest.repos.getContent({
+                        owner, repo, path: theme.screenshot
+                    });
+                    const imageMeta = await probe(screenshot.data.download_url);
+                    if (imageMeta.type !== 'png' && imageMeta.type !== 'jpg') {
+                        addError('Theme screenshot is not of filetype `.png` or `.jpg`');
+                    }
+                    const recommendedSize = 'we generally recommend a size around 512 × 288 pixels';                    
+                    if (imageMeta.width > 1000 || imageMeta.height > 500) {
+                        addError(`Your theme screenshot is too big, ${recommendedSize}`);
+                    }
+                    if (imageMeta.width < 250 || imageMeta.height < 100) {
+                        addError(`Your theme screenshot is too small, ${recommendedSize}`);
+                    }
```

Comment:
I would add a warning, if neither is the case, to tell them what the recommended size is. Also these statements should be `else if` rather than two `if`s since they can collide if they submit a really wide or long screenshot - then you'd say it's too big and too small.